executor/aggfuncs/func_cume_dist.go

```diff
@@ -42,14 +43,28 @@ func (r *cumeDist) ResetPartialResult(pr PartialResult) {
 
 func (r *cumeDist) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) error {
 	p := (*partialResult4CumeDist)(pr)
-	p.rows = append(p.rows, rowsInGroup...)
+	for _, row := range rowsInGroup {
+		rowDatum := make([]*types.Datum, len(r.colIdx))
+		for i, idx := range r.colIdx {
+			d := row.GetDatum(idx, r.retTypes[i])
```

Comment:
If the datum type is string, I think it is still an reference.

Reply:
PTAL again.